Design/Instrument-2.0/ScanningInstruments_Design.md

```diff
@@ -18,7 +18,20 @@ Two classes related to this have already been partially implemented in Mantid, [
 
 Support for the moving instruments will be implemented via the `DetectorInfo` class. The usage `SpectrumInfo` class needs no direct knowledge of the time indexing. The methods on the `DetectorInfo` class for accessing detector information will be accessed by giving a detector index and an optional time index. It will also hold information on the start and end times for each time index. Any further information, for example to normalise to monitor counts, can be held in a time series sample log.
 
-Currently initialisation of workspaces is usually done in an incremental way. This will have performance impacts for the loader, but can also leave the workspace in an inconsistent state. For the scanning instrument design the workspace should be initialised all at once, via a `DetectorInfo` factory, which will ensure the loader creates a workspace in a consistent state.
+Currently initialisation of workspaces is usually done in an incremental way. Exactly how this is done varies between loaders, but is often spectrum-by-spectrum. For example a common pattern for an instrument with PSD tubes would be as follows:
+
+```cpp
+for (size_t i = 0; i < numberOfTubes; ++i) {
+  for (size_t j = 0; j < m_numberOfPixelsPerTube; ++j) {
+    int *data_p = &data(static_cast<int>(i), static_cast<int>(j), 0);
+    m_localWorkspace->setHistogram(spec, m_localWorkspace->binEdges(0),
+        Counts(data_p, data_p + m_numberOfChannels));
+    spec++;
+  }
+}
+```
+
+This will have performance impacts for the loader, but can also leave the workspace in an inconsistent state. For the scanning instrument design the workspace should be initialised all at once, via a `DetectorInfo` factory, which will ensure the loader creates a workspace in a consistent state.
```

Comment:
You might want to make a point of saying that the direct analogue of the problem you provided the code for above is the population of information relevant to the geometry caching `DetectorInfo` layer, such as `setPosition`. Your example above is good, and you should keep it, but it relates to the `Histogram` aspect of the workspace rather than anything that would go in the `DetectorInfo` and that needs to be made clear to avoid confusion.